src/sentry/api/endpoints/sentry_app_components.py

```
@@ -27,10 +27,12 @@ def get(self, request, sentry_app):
 class OrganizationSentryAppComponentsEndpoint(OrganizationEndpoint):
     @add_integration_platform_metric_tag
     def get(self, request, organization):
+        project_id = request.GET.get("projectId")
+        if not project_id:
+            return Response([], status=404)
```

@mgaeta we should return a validation error about the project being missing